backend/capellacollab/sessions/routes.py

```diff
@@ -87,70 +96,66 @@ def get_current_sessions(
 
 
 @router.post(
-    "/",
+    "/readonly",
     response_model=AdvancedSessionResponse,
 )
 def request_session(
-    body: PostSessionRequest,
+    body: PostReadonlySessionRequest,
     db_user: DatabaseUser = Depends(get_own_user),
     operator: Operator = Depends(get_operator),
     db: Session = Depends(get_db),
     token=Depends(JWTBearer()),
 ):
-    assert body.type == WorkspaceType.READONLY
-
     rdp_password = generate_password(length=64)
 
-    owner = db_user.username
+    owner = db_user.name
 
     log.info("Starting persistent session creation for user %s", owner)
 
     existing_user_sessions = database.get_sessions_for_user(db, owner)
 
-    if body.repository in [
-        session.repository for session in existing_user_sessions
-    ]:
+    project = get_project_by_slug(db, body.project_slug)
+    if not project:
         raise HTTPException(
-            status_code=404,
-            detail={
-                "err_code": "existing_session",
-                "reason": f"You already have a open Read-Only Session for the repository {body.repository}. Please navigate to 'Active Sessions' to Reconnect",
+            404,
+            {
+                "reason": f"The project with name {body.project_slug} was not found.",
+                "technical": f"No project {body.project_slug} found.",
             },
         )
-    verify_project_role(repository=body.repository, token=token, db=db)
-    git_model = git_models_crud.get_primary_model_of_repository(
-        db, body.repository
-    )
-    if not git_model:
+
+    verify_project_role(project.name, token, db)
```

Comment:
If you rebased on _v2/staging_, you should rather use the injectables to check the permissions.

Reply:
I tried that, but it made my tests fail. I'd like to leave it as is for now, and have a look at improving the code in this area later.